File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Program/InviteDto.java
package com.salesianos.triana.BeElite.dto.Program;

import com.salesianos.triana.BeElite.model.InvitationStatus;
import com.salesianos.triana.BeElite.model.Invite;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record InviteDto(UUID inviteId,
                        LocalDateTime createdAt,
                        String coachName,
                        String programName,
                        String athleteUsername,
                        String status) {

    public static InviteDto of(Invite i){
        return InviteDto.builder()
                .inviteId(i.getId())
                .createdAt(i.getCreatedAt())
                .coachName(i.getProgram().getCoach().getName())
                .programName(i.getProgram().getProgramName())
                .athleteUsername(i.getAthlete().getUsername())
                .status(i.getStatus())
                .build();
    }
}

File: frontend/be_elite/lib/models/Program/invite_dto.dart
class InviteDto {
  String? inviteId;
  DateTime? createdAt;
  String? coachName;
  String? programName;
  String? athleteUsername;
  String? status;

  InviteDto({
    this.inviteId,
    this.createdAt,
    this.coachName,
    this.programName,
    this.athleteUsername,
    this.status,
  });

  factory InviteDto.fromJson(Map<String, dynamic> json) => InviteDto(
        inviteId: json['inviteId'] as String?,
        createdAt: json['createdAt'] == null
            ? null
            : DateTime.parse(json['createdAt'] as String),
        coachName: json['coachName'] as String?,
        programName: json['programName'] as String?,
        athleteUsername: json['athleteUsername'] as String?,
        status: json['status'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'inviteId': inviteId,
        'createdAt': createdAt?.toIso8601String(),
        'coachName': coachName,
        'programName': programName,
        'athleteUsername': athleteUsername,
        'status': status,
      };
}
